keanu-python/keanu/stats.py

```diff
@@ -0,0 +1,21 @@
+from typing import List
+
+from numpy import ndarray, fromiter, issubdtype, hstack, floating
+from py4j.java_gateway import java_import
+
+from keanu.vartypes import numpy_types
+from .context import KeanuContext
+
+k = KeanuContext()
+
+java_import(k.jvm_view(), "io.improbable.keanu.algorithms.SampleStats")
+
+
+def autocorrelation(arg: List[numpy_types]) -> ndarray:
+    if not issubdtype(arg[0].dtype, floating):
+        raise ValueError('Autocorrelation must be run on a list of numpy floating types.')
+    if not arg[0].shape == ():
```

Comment:
This seems too restrictive. Maybe we should do something like allowing the user to specify the index instead (with a sensible default)?

Reply:
This is a good point, something I completely missed. I am working on adding an argument for the index, which will make it very similar to how it's done in Java.